ewatercycle/models/lisflood.py

```diff
@@ -0,0 +1,398 @@
+        stdout, stderr = container.communicate()
+        return exit_code, stdout, stderr
+
+    def get_value_as_xarray(self, name: str) -> xr.DataArray:
+        """Return the value as xarray object."""
+        # Get time information
+        time_units = self.bmi.get_time_units()
+        grid = self.bmi.get_var_grid(name)
+        shape = self.bmi.get_grid_shape(grid)
+
+        # Extract the data and store it in an xarray DataArray
+        da = xr.DataArray(
+            data=np.reshape(self.bmi.get_value(name), shape),
+            coords={
+                "longitude": self.bmi.get_grid_y(grid),
+                "latitude": self.bmi.get_grid_x(grid),
+                "time": num2date(self.bmi.get_current_time(), time_units)
+            },
+            dims=["latitude", "longitude"],
+            name=name,
+            attrs={"units": self.bmi.get_var_units(name)},
+        )
+
+        return da
+
+    @property
+    def parameters(self) -> Iterable[Tuple[str, Any]]:
+        """List the parameters for this model."""
+        if self.forcing_dir and self.work_dir:
+            return {
+                'forcing_dir': self.forcing_dir,
+                'work_dir': self.work_dir,
+            }.items()
+        return []
+
+
+def reindex_forcings(mask_map: PathLike, forcing: ForcingData, output_dir: PathLike = None) -> PathLike:
+    """Reindex forcing files to match mask map grid
+
+    Args:
+        mask_map: Path to NetCDF file used a boolean map that defines model boundaries.
+        forcing: Forcing data from ESMValTool
+        output_dir: Directory where to write the re-indexed files, given by user or created for user
+
+    Returns:
+        Output dir with re-indexed files.
+    """
+    output_dir = _generate_workdir(output_dir)
+    mask = xr.open_dataarray(mask_map).load()
+    data_files = list(forcing.recipe_output.values())[0].data_files
+    for data_file in data_files:
+        dataset = data_file.load_xarray()
+        out_fn = output_dir / data_file.filename.name
+        var_name = list(dataset.data_vars.keys())[0]
+        encoding = {var_name: {"zlib": True, "complevel": 4, "chunksizes": (1,) + dataset[var_name].shape[1:]}}
+        dataset.reindex(
+                    {"lat": mask["lat"], "lon": mask["lon"]},
+                    method="nearest",
+                    tolerance=1e-2,
+                ).to_netcdf(out_fn, encoding=encoding)
+    return output_dir
+
+
+def _generate_workdir(work_dir: PathLike = None) -> PathLike:
+    """
+
+    Args:
+        work_dir: If work dir is None then create sub-directory in CFG['output_dir']
+
+    """
+    if work_dir is None:
+        scratch_dir = CFG['output_dir']
+        # TODO this timestamp isnot safe for parallel processing
+        timestamp = time.strftime("%Y%m%d_%H%M%S")
+        work_dir = Path(scratch_dir) / f'lisflood_{timestamp}'
+        work_dir.mkdir(parents=True, exist_ok=True)
+    return work_dir
+
+
+class XmlConfig(AbstractConfig):
```

Comment:
Should we move this to `ewatercycle/parametersetdb/config` so it can be reused by other (future) model classes as well?

Reply:
We (@sverhoeven and I) discussed this. `source` is a path in this class whereas it is an URL in other classes in `ewatercycle/parametersetdb/config`.

Reply:
Oh right, yes, I stumbled upon that as well. Perhaps we can try to make that more flexible in `parametersetdb`?

Reply:
The LisFlood model is the only model that uses XML, when we add a another model that uses XML we can move it over

Reply:
Created an issue: https://github.com/eWaterCycle/ewatercycle/issues/58